pkg/dockerfile/fast_generator.go

```diff
@@ -0,0 +1,273 @@
+	return nil
+}
+
+func (g *FastGenerator) GenerateDockerfileWithoutSeparateWeights() (string, error) {
+	return g.generate()
+}
+
+func (g *FastGenerator) GenerateModelBase() (string, error) {
+	return "", errors.New("GenerateModelBase not supported in FastGenerator")
+}
+
+func (g *FastGenerator) GenerateModelBaseWithSeparateWeights(imageName string) (weightsBase string, dockerfile string, dockerignoreContents string, err error) {
+	return "", "", "", errors.New("GenerateModelBaseWithSeparateWeights not supported in FastGenerator")
+}
+
+func (g *FastGenerator) GenerateWeightsManifest() (*weights.Manifest, error) {
+	return nil, errors.New("GenerateWeightsManifest not supported in FastGenerator")
+}
+
+func (g *FastGenerator) IsUsingCogBaseImage() bool {
+	return false
+}
+
+func (g *FastGenerator) SetPrecompile(precompile bool) {
+}
+
+func (g *FastGenerator) SetStrip(strip bool) {
+}
+
+func (g *FastGenerator) SetUseCogBaseImage(useCogBaseImage bool) {
+}
+
+func (g *FastGenerator) SetUseCogBaseImagePtr(useCogBaseImage *bool) {
+}
+
+func (g *FastGenerator) SetUseCudaBaseImage(argumentValue string) {
+}
+
+func (g *FastGenerator) Name() string {
+	return FAST_GENERATOR_NAME
+}
+
+func (g *FastGenerator) generate() (string, error) {
+	tmpDir, err := BuildCogTempDir(g.Dir)
+	if err != nil {
+		return "", err
+	}
+
+	weights, err := FindWeights(g.Dir, tmpDir)
+	if err != nil {
+		return "", err
+	}
+
+	lines := []string{}
+	lines, err = g.generateMonobase(lines, tmpDir)
+	if err != nil {
+		return "", err
+	}
+
+	lines, err = g.copyWeights(lines, weights)
+	if err != nil {
+		return "", err
+	}
+
+	lines, err = g.install(lines, weights, tmpDir)
+	if err != nil {
+		return "", err
+	}
+
+	lines, err = g.entrypoint(lines)
+	if err != nil {
+		return "", err
+	}
+
+	return strings.Join(lines, "\n"), nil
+}
+
+func (g *FastGenerator) copyCog(tmpDir string) (string, error) {
+	files, err := CogEmbed.ReadDir("embed")
+	if err != nil {
+		return "", err
+	}
+	if len(files) != 1 {
+		return "", fmt.Errorf("should only have one cog wheel embedded")
+	}
+	filename := files[0].Name()
+	data, err := CogEmbed.ReadFile("embed/" + filename)
+	if err != nil {
+		return "", err
+	}
+	path := filepath.Join(tmpDir, filename)
+	if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
+		return "", fmt.Errorf("Failed to write %s: %w", filename, err)
+	}
+	if err := os.WriteFile(path, data, 0o644); err != nil {
+		return "", fmt.Errorf("Failed to write %s: %w", filename, err)
+	}
+	return path, nil
+}
+
+func (g *FastGenerator) generateMonobase(lines []string, tmpDir string) ([]string, error) {
+	lines = append(lines, []string{
+		"# syntax=docker/dockerfile:1-labs",
+		"FROM r8.im/monobase:latest",
+	}...)
+
+	cogPath, err := g.copyCog(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+
+	lines = append(lines, []string{
+		"ENV R8_COG_VERSION=\"file:///buildtmp/" + filepath.Base(cogPath) + "\"",
+	}...)
+
+	skipCudaArg := "--skip-cuda"
```

Comment:
We should get rid of `--skip-cuda` everywhere. It's to speed up CI only and I renamed to `CI_SKIP_CUDA` to reduce confusion.

Reply:
Resolved in [9dbbe5a](https://github.com/replicate/cog/pull/2108/commits/9dbbe5a3c184061c20d71397e2bf740aa76b7e9d)